core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1025,7 +991,51 @@ class GroupCoordinator(val brokerId: Int,
     } else {
       group.removePendingMember(memberId)
     }
-    maybePrepareRebalance(group, s"Adding new member $memberId with group instanceid $groupInstanceId")
+    maybePrepareRebalance(group, s"Adding new member $memberId with group instance id $groupInstanceId")
+  }
+
+  private def updateStaticMemberAndRebalance(group: GroupMetadata,
```

Comment:
The logic we have in `doSyncGroup` to fill in missing assignments seems dubious. Perhaps there's a compatibility argument to retain it, but it seems more likely to cause problems like the one here than to be useful.